File: server/Models/Keep.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Keep : RepoItem<int>
{
  [MaxLength(100)] public string Name { get; set; }
  [MaxLength(3000)] public string Description { get; set; }
  [Url, MaxLength(1000)] public string Img { get; set; }
  public int Views { get; set; }
  public string CreatorId { get; set; }
}

File: client/src/models/Keep.js
import { Profile } from "./Account.js"

export class Keep {
  constructor(data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.description = data.description
    this.creator = new Profile(data.creator)
    this.creatorId = data.creatorId
    this.img = data.img
    this.kept = data.kept
    this.name = data.name
    this.views = data.views
    this.vaultKeepId = data.vaultKeepId
  }
}